File: src/DummyTask.java
import java.util.Random;

/**
 * A simple task to test our genetic optimization framework.
 * The goal is to "guess" a vector of 5 values.
 */
public class DummyTask implements Solution {

    double[] coordinates;
    double[] goal = new double[] {1, 2, 3, 4, 5};
    Random rng;

    public DummyTask(double[] elements){
       coordinates = new double[5];
       for (int i=0; i<5; i++){
           coordinates[i] = elements[i];
       }
       rng = new Random();
    }

    // A random coordinate is modified slightly.
    public DummyTask Mutate(double rate) {
        DummyTask result = new DummyTask(coordinates);
        int i = rng.nextInt(5);
        result.coordinates[i] += (rng.nextDouble()-0.5)*rate*coordinates[i];
        return result;
    }

    // Each coordinate is chosen from a random parent.
    public DummyTask MateWith(Solution partner) {
        DummyTask result = new DummyTask(coordinates);
        for (int i=0; i<5; i++){
            if (rng.nextBoolean()){
                result.coordinates[i] = ((DummyTask)partner).coordinates[i];
            }
        }
        return result;
    }

    // sum of squared differences between the coordinates
    public double Fitness() {
        double diff = 0;
        for (int i=0; i<5; i++){
            double d = coordinates[i]-goal[i];
            diff += d*d;
        }
        return -diff;
    }
}

File: src/python/dummytask.py
from solution import Solution
from random

class DummyTask(Solution):
    def __init__(self, elements):
        super(Solution, self).__init__()

        ## custom init after interface init
        self.coordinates = elements
        self.goal = [1, 2, 3, 4, 5]

    def mutate(self, rate):
        result = DummyTask(self.coordinates)
        i = random.randint(1,5)
        resut.coordinates[i] += (random.random() - 0.5) * rate * self.coordinates[i]
        return result

    def mateWith(self, partner):
        reuslt = DummyTask(self.coordinates)
        result = [result[i] if random.random() < 0.5 else partner.coordinates[i] for i in range(5)]
        return [result]

    def fitness(self):
        return -sum([(self.coordinates[i] - self.goal[i]) * (self.coordinates[i] - self.goal[i])] for i in range(5))
